relayer/src/connection.rs

```diff
@@ -500,114 +410,231 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Connection<ChainA, ChainB> {
         }
     }
 
-    /// Executes a connection handshake protocol (ICS 003) for this connection object
-    fn handshake(&mut self) -> Result<(), ConnectionError> {
-        let done = '🥂';
-
-        let a_chain = self.a_side.chain.clone();
-        let b_chain = self.b_side.chain.clone();
-
-        // Try connOpenInit on a_chain
-        let mut counter = 0;
-        while counter < MAX_RETRIES {
-            counter += 1;
-            match self.flipped().build_conn_init_and_send() {
-                Err(e) => {
-                    error!("failed ConnInit {:?}: {}", self.a_side, e);
-                    continue;
-                }
-                Ok(result) => {
-                    self.a_side.connection_id = Some(extract_connection_id(&result)?.clone());
-                    println!("🥂  {} => {:#?}\n", self.a_side.chain.id(), result);
-                    break;
-                }
-            }
+    /// Queries the chains for latest connection end information. It verifies the relayer connection
+    /// IDs and updates them if needed.
+    /// Returns the states of the two connection ends.
+    ///
+    /// The relayer connection stores the connection identifiers on the two chains a and b.
```

Comment:
Subtle question here: are the connection identifiers stored _on_ the chains A and B, or are they stored on the relayer and _pertain_ to chains A and B?

I ask because the next sentence seems to indicate that there are corresponding on-chain identifiers, which makes me think that the latter statement is correct.

Reply:
Not sure I understand the question. The IDs of all IBC objects (clients, connections, channels) are assigned by the on-chain handlers at creation time and stored on-chain. The relayer gets them from the response to the message that created them. In the connection handshake case, the relayer saves them in the `Connection` object under `side_a` or `side_b`.

Reply:
Ok, that makes sense :+1: